build/prefilter.ml

```diff
@@ -37,6 +37,12 @@ let print_loc = function
        stale := false;
      end
 
+let has_domains ~extra =
+  let extra = String.split_on_char '+' extra in
+  List.mem "domains" extra ||
+  List.mem "multicore" extra ||
+  List.mem "effects" extra
```

Comment:
This may have to be refined again in the future, given that `effects` may add extra stdlib logic in addition to just `domains`. But I'm okay with going for this for now, and refining as it comes in the future.